cores/esp32/esp32-hal-adc.c

```diff
@@ -310,4 +336,357 @@ extern void analogSetPinAttenuation(uint8_t pin, adc_attenuation_t attenuation)
+    err = adc_continuous_register_event_callbacks(adc_handle[adc_unit].adc_continuous_handle, &cbs, &adc_handle[adc_unit].adc_interrupt_handle);
+    if(err != ESP_OK){
+        log_e("adc_continuous_register_event_callbacks failed!");
+        return ESP_FAIL;
+    }
+
+    //Allocate and prepare result structure for adc readings
+    adc_result = malloc(pins_count * sizeof(adc_continuos_data_t));
+    for(int k = 0; k < pins_count; k++){
+        adc_result[k].pin = pins[k];
+        adc_result[k].channel = channel[k];
+    }
+
+    //Initialize ADC calibration handle
+    if(adc_handle[adc_unit].adc_cali_handle == NULL){
+        log_d("Creating cali handle for ADC_%d", adc_unit);
+        #if ADC_CALI_SCHEME_CURVE_FITTING_SUPPORTED
+            adc_cali_curve_fitting_config_t cali_config = {
+                .unit_id = adc_unit,
+                .atten = __adcContinuousAtten,
+                .bitwidth = __adcContinuousWidth,
+            };
+            err = adc_cali_create_scheme_curve_fitting(&cali_config, &adc_handle[adc_unit].adc_cali_handle);
+        #elif !defined(CONFIG_IDF_TARGET_ESP32H2) //ADC_CALI_SCHEME_LINE_FITTING_SUPPORTED
+            adc_cali_line_fitting_config_t cali_config = {
+                .unit_id = adc_unit,
+                .bitwidth = __adcContinuousWidth,
+                .atten = __adcContinuousAtten,
+            };
+            err = adc_cali_create_scheme_line_fitting(&cali_config, &adc_handle[adc_unit].adc_cali_handle);
+        #endif
+        if(err != ESP_OK){
+            log_e("adc_cali_create_scheme_x failed!");
+            return ESP_FAIL;
+        }
+    }
+
+    for(int k = 0; k < pins_count; k++){
+        if(!perimanSetPinBus(pins[k], ESP32_BUS_TYPE_ADC_CONT, (void *)(adc_unit+1))){
+            log_e("perimanSetPinBus to ADC Continuous failed!");
+            adcContinuousDetachBus((void *)(adc_unit+1));
+            return ESP_FAIL;
+        }
+    }
+
+    return ESP_OK;
+}
+
+bool analogContinuousRead(adc_continuos_data_t ** buffer, uint32_t timeout_ms){
+    if(adc_handle[ADC_UNIT_1].adc_continuous_handle != NULL){
+        uint32_t bytes_read = 0;
+        uint32_t read_raw[used_adc_channels];
+        uint32_t read_count[used_adc_channels];
+        uint8_t adc_read[adc_handle[ADC_UNIT_1].conversion_frame_size];
+        memset(adc_read, 0xcc, sizeof(adc_read));
+        memset(read_raw, 0, sizeof(read_raw));
+        memset(read_count, 0, sizeof(read_count));
+
+        esp_err_t err = adc_continuous_read(adc_handle[ADC_UNIT_1].adc_continuous_handle, adc_read, adc_handle[0].conversion_frame_size, &bytes_read, timeout_ms);
+        if(err != ESP_OK){
+            if(err == ESP_ERR_TIMEOUT){
+                log_e("Reading data failed: No data, increase timeout");
+            }
+            else {
+                log_e("Reading data failed with error: %X", err);
+            }
+            *buffer = NULL;
+            return ESP_FAIL;
```

Comment:
functions marked as returning bool, but you are returning esp_err_t in them?

Reply:
@me-no-dev will it be better for users to return bool or esp_err_t?

Reply:
bool is OK for Arduino